Replaced Traffic Manager with auto-generated version and implemented new API for it

number213 opened the pull request:
* Replaced Traffic Manager with autogenerated version
* Implemented new API for Traffic Manager (Fast Endpoint Failover)

Node SDK PR with new API: https://github.com/Azure/azure-sdk-for-node/pull/2158

The content to be added to Changelog is as follows:
* Network
	* Traffic Manager
		* Rewrote implementation for Traffic Manager
		* Added support for API version 2017-05-01 (Fast Endpoint Failover)

balajikris commented:
hey @number213 - are you (or the team that owns this pkg) planning to publish the latest version of the sdk using this [link](http://azuresdkci.cloudapp.net/view/5-Signing%20and%20Publishing/job/sdk-for-node-publish/). until that updated package is in npm, this cannot be merged.

number213 commented:
@balajikris I have no access to Jenkins, but I've got a member of our team who offered to publish the package a couple of days before code freeze. I'll ping you once we are ready.

number213 commented:
Hi @balajikris, PR is ready for review

vladca commented on lib/commands/arm/network/TrafficManagers-profiles._js:
```diff
@@ -0,0 +1,497 @@
+        parameters.monitorConfig.port = parseInt(defaultPort, 10);
+      }
+
+      if(options.monitorPath) {
```
In the REST API, Path is now mandatory if Protocol=HTTP/HTTPS and Path must be null if Protocol=TCP; @ms-alguerra

number213 replied:
Conditional requirement isn't support by code generator right now, but we can add it if client-side validation is required..

vladca commented on lib/commands/arm/network/TrafficManagers-profiles._js:
```diff
@@ -0,0 +1,497 @@
+      var profileResultVar = null;
+      var progress;
+      try {
+        if(typeof trafficManagerManagementClient.profiles.listAll != 'function') {
```
What else can it be if not function?

number213 replied:
It could be `undefined`.
This is universal check we used for Network because some commands could list all entities without specific resource group, but some couldn't.

vladca commented:
Client side validation is not required, but the client must not prevent any of these cases

number213 commented:
Hi, @vladca, @ms-alguerra, I update PRю
Removed using `/` as default Path and added removal of Path param if protocol passed is `TCP`.
Could you please review this PR again?

number213 commented:
Hi, @vladca, @ms-alguerra, I updated PR again, could you please review it?

balajikris commented:
Hey Guys, 
Just checking in to see what's the status of this PR? It seems like we're waiting on @vladca @ms-alguerra to take a look ? Please note that June 7 is the code freeze for the next release, as seen here: https://github.com/Azure/azure-xplat-cli/milestone/37

vladca commented:
I am happy with this PR, but I am not a repo owner so I cannot merge it.

balajikris commented:
Thanks @vladca - I just wanted to know if you were okay with it. I've merged it.
